core/domain/state_domain.py

```diff
@@ -2785,21 +2785,17 @@ def update_interaction_answer_groups(self, answer_groups_list):
 
         # TODO(yanamal): Do additional calculations here to get the
         # parameter changes, if necessary.
-        for answer_group_dict in answer_groups_list:
-            rule_specs_list = answer_group_dict['rule_specs']
+        for answer_group in answer_groups_list:
+            rule_specs_list = answer_group.rule_specs
             if not isinstance(rule_specs_list, list):
                 raise Exception(
                     'Expected answer group rule specs to be a list, '
                     'received %s' % rule_specs_list)
 
-            answer_group = AnswerGroup(
-                Outcome.from_dict(answer_group_dict['outcome']), [],
-                answer_group_dict['training_data'],
-                answer_group_dict['tagged_skill_misconception_id'])
+            answer_group.rule_specs = []
```

Comment:
Why this change? It seems like this field is not used anywhere else in this function, does it need to be empty?

Reply:
Earlier it is initilizing answer_group by forming an object with answer_group_dict and the only different attribute is rule_spec which is an empty list.
Currently answer_group_dict is replaced with the variable name as answer_group, so only update needed is to initialize it with an empty list.
Also answer_group is used in line 2804.
Hope I am able to explain my point, Is I'm missing something ?